auth/store.go

```diff
@@ -40,12 +40,16 @@ var (
 	ErrUserNotFound     = errors.New("auth: user not found")
 	ErrRoleAlreadyExist = errors.New("auth: role already exists")
 	ErrRoleNotFound     = errors.New("auth: role not found")
+	ErrAuthFailed       = errors.New("auth: authentication failed, invalid user ID or password")
 )
 
 type AuthStore interface {
 	// AuthEnable() turns on the authentication feature
 	AuthEnable()
 
+	// Authenticate() does authentication based on given user name and password
+	Authenticate(name string, password string) (*pb.AuthenticateResponse, error)
```

Comment:
we should mention that the returned token is only valid for the member it connected to within x minutes. if it is not used for x minutes, it will be removed and reauth is needed. (at least for now for the simple token, we can improve this later)